scipy/stats/_multivariate.py

```diff
@@ -839,6 +879,10 @@ def __init__(self, mean=None, cov=1, allow_singular=False, seed=None,
         self.abseps = abseps
         self.releps = releps
 
+    @property
+    def cov(self):
+        return self.cov_object.covariance
+
```

Comment:
I think that this should return whatever the user passed in as `cov`. 
- If the user passed in a matrix (as a list or np.array), then `self.cov` will be an object of type `CovViaPSD` (which is not a public class), and we should return `self.cov_object.covariance`. 
- If the user passed in a `Covariance` object for cov, then that is stored directly in `self.cov`, and this should return `self.cov`

Reply:
I think that's a pretty complicated set of semantics, and doesn't really fix one of the backwards compatibility issues. Because the old interface was only semi-documented, what the `.cov` attribute guaranteed was a little fuzzy, but it wasn't "whatever was passed in as `cov`". The input did get processed and normalized to a full covariance matrix from a scalar or a diagonal or a list of lists, etc. into a 2-dim array that was the covariance matrix. So someone could well have written library functions that take `multivariate_normal_frozen` instances and access the `.cov` to get the fully normalized covariance matrix. I think it makes sense to honor those semantics here.

Reply:
Sure. Whatever code was written before 1.10 will still work with what I suggested.

Yes, "whatever the user passed in as `cov`" is not very accurate. I forgot that there are over 50 lines of processing to convert the user input into a standard-format ndarray. It would be difficult to capture all of that in the docstring.

Perhaps if it is to become public it could be described as:
```
    cov : ndarray or Covariance
        Covariance of the distribution.
```

Reply:
Whole programs will still work (ones that create the `multivariate_normal_frozen` instances then access their `.cov` attribute), but not necessarily individual functions that were written against a (very reasonable) guess at what was being promised by the interface. Both are aspects of backwards compatibility that would have to be restored if we really want to care about backwards compatibility here, not just whole-program compatibility.

If you really don't want these extra attributes (a very reasonable design decision), just hide them behind deprecated properties and move on (but make sure that the `cov` property returns the actual array along with the `DeprecationWarning`). There probably are very few programs that ever relied on these.

Reply:
If I understand correctly, this is not backward compatible with all code written before 1.10
```python3
    @property
    def cov(self):  # this is what I had in mind
        if isinstance(self.cov_object, CovViaPSD)
            return self.cov_object.covariance
        else:  # cov_object is a different type of Covariance
            return self.cov_object
```

but this is:
```python3
    @property
    def cov(self):  # this is what is in the PR
        return self.cov_object.covariance
```

But I'm having trouble understanding why. If it's simple to provide an example that might have been written before 1.10 which would be broken by the former but not the latter, I'd appreciate it.

Reply:
I found another mistake in my comment at the top; perhaps this will help. In the bulleted list when I wrote `self.cov`, I meant `self.cov_object`.

- If the user passed in a matrix (as a list or np.array), then `self.cov_object` will be an object of type `CovViaPSD` (which is not a public class), and we should return `self.cov_object.covariance`. 
- If the user passed in a `Covariance` object for cov, then that is stored directly in `self.cov_object`, and this should return `self.cov_object`

Reply:
Since this feature was not really documented before, couldn't this be the time to make the decision to have `self.cov` consistently return the matrix representation of the covariance?

This was what I had in mind, it may not be completely backwards compatible with version 1.10.0 since in that one the user may have passed in a `Covariance` object and expected it back, but that's been out for a relatively short time so it seems acceptable to me (correct me if I'm wrong about this of course).

Reply:
Maybe this is the explanation I needed:

- Before SciPy 1.10, library writes function `f` that accepts a `multivariate_normal_frozen` as an input `dist`. If `getattr(dist, 'cov', None)` does not produce an array, `f` will fail.
- After SciPy 1.10 is released, user writes new code that creates `dist` with `Covariance` object, calls library's function as `f(dist)`. In `f`, `getattr(dist, 'cov', None)` returns a `Covariance` object instead of an array, and therefore `f` fails.

(Incorrect perspective: User reports to library that it does not work with `dist` objects created with instance of `Covariance`. Library fixes bug.)

Reply:
If you want to expose the `Covariance` object underlying the `multivariate_normal_frozen` instance, exposing the `cov_object` is the best way to do that, not this variable behavior with the `cov` attribute. Having `.cov` be either an array or a `Covariance` object depending on how it was created is a very difficult to use API. Functions that consume these instances should be able to be written in such a way that does not care about that detail because those functions are far away from the instantiation of those objects.

If you don't want to expose the `Covariance` object or the explicit covariance matrix, just go with the PR's implementation of the property, deprecate it, and remove it at the scheduled time.